csrc/ir/nodes.cpp

```diff
@@ -3650,6 +3650,61 @@ void TensorDomain::setAllocationDomain(
   contiguity_ = std::move(new_contiguity);
 }
 
+std::vector<IterDomain*> TensorDomain::allIDs() const {
+  std::array<const std::vector<IterDomain*>*, 4> all_domains = {
```

Comment:
nit: these four sets of domains would commonly have some redundancy. For example, root and logical would be the same for many tensors. Doing some filtering seems like a quick optimization so that we won't do `getExprsBetween` with `logical` and `allocation`, and also `root` and `allocation`.

Reply:
Isn't the
```C++
      if (all_domains[j]->empty()) {
        continue;
      }
```
sufficient?

Reply:
You're right. Missed that.